test(ui): E2E Test - New user sign-up/registration

StylusFrost opened the pull request:
### Context

This PR implements the critical path of new user sign-up/registration e2e test flow. PR DEPENDECY https://github.com/prowler-cloud/prowler/pull/8881

### Description
The implementation includes a comprehensive E2E test suite (SIGNUP-E2E-001) that validates the complete user registration flow:

- Navigate to Sign up page

- Fill form with valid data (unique email, valid password, terms accepted)

- Submit form and verify redirect to Login page

- Confirm newly created credentials can be used for successful authentication

- Verify user lands on home page with main content visible

### Steps to review

```
cd ui
npx playwright test --grep @SIGNUP-E2E-001
```

### Checklist

- Are there new checks included in this PR?  No
    - If so, do we need to update permissions for the provider? Please review this carefully.
- [ ] Review if the code is being covered by tests.
- [ ] Review if code is being documented following this specification https://github.com/google/styleguide/blob/gh-pages/pyguide.md#38-comments-and-docstrings
- [ ] Review if backport is needed.
- [ ] Review if is needed to change the [Readme.md](https://github.com/prowler-cloud/prowler/blob/master/README.md)
- [ ] Ensure new entries are added to [CHANGELOG.md](https://github.com/prowler-cloud/prowler/blob/master/prowler/CHANGELOG.md), if applicable.

#### API
- [ ] Verify if API specs need to be regenerated.
- [ ] Check if version updates are required (e.g., specs, Poetry, etc.).
- [ ] Ensure new entries are added to [CHANGELOG.md](https://github.com/prowler-cloud/prowler/blob/master/api/CHANGELOG.md), if applicable.

### License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

StylusFrost committed:
test(ui): enhance Playwright test setups for user authentication

- Added multiple authentication setup files for different user roles (admin, manage scans, manage integrations, etc.) to streamline end-to-end testing.
- Introduced a helper function to authenticate users and save their state for reuse in tests.
- Updated Playwright configuration to include new authentication projects.

This change improves the testing framework by allowing for more comprehensive and role-specific user authentication scenarios.

StylusFrost committed:
test(ui): update Playwright test commands to specify project. Compatibility with current e2e test

- Modified Playwright test commands in package.json to explicitly use the 'chromium' project.

StylusFrost committed:
test(ui): format Playwright configuration for consistency

- Added missing commas and improved formatting in the Playwright configuration file to enhance readability and maintain consistency across the codebase.

StylusFrost committed:
test(ui): implement base page object and enhance sign-up flow tests

- Introduced a BasePage class to encapsulate common functionality for page objects, improving code reusability and maintainability.
- Created new page objects for HomePage, SignInPage, and SignUpPage to streamline the sign-up and login processes in tests.
- Added comprehensive sign-up flow tests to validate user registration and login functionality, ensuring a smooth user experience.
- Updated Playwright configuration to support new test structures and improve overall test organization.

StylusFrost committed:
test(ui): add E2E tests document for user sign-up flow

- Documented test case details including flow steps, expected results, and key verification points.
- Enhanced existing sign-up test specifications to include relevant tags for better categorization and tracking.

StylusFrost committed:
test(ui): simplify sign-up page tests by removing redundant loading state verification

- Removed the `verifyLoadingState` method from the `SignUpPage` class as it was redundant.
- Updated comments in the sign-up test to enhance clarity and focus on key actions.
- Added a call to `verifyNoErrors` to ensure no errors occur during the sign-up process.

StylusFrost committed:
test(ui): enhance session management tests with new helper functions

- Updated the `verifyLogoutSuccess` function to use a regex for URL verification.
- Added new helper functions `getSession` and `verifySessionValid` to streamline session validation in tests, ensuring that session data is correctly retrieved and validated.

StylusFrost committed:
Merge branch 'master' into PROWLER-187-create-new-user

StylusFrost committed:
test(ui): add page load wait to sign-up page tests

- Introduced a call to `waitForPageLoad` in the sign-up page test to ensure the page is fully loaded before proceeding with visibility checks for elements.

StylusFrost committed:
Merge branch 'master' into PROWLER-187-create-new-user

alejandrobailo commented on ui/tests/sign-up/sign-up.spec.ts:
```diff
@@ -0,0 +1,41 @@
+      company: `Test E2E Co ${suffix}`,
+      email: uniqueEmail,
+      password: "Thisisapassword123@",
+      confirmPassword: "Thisisapassword123@",
```
Let's use .env keys

alejandrobailo reviewed (changes requested):
Missing these error cases: 
- Duplicate email
- Password mismatch
- Required fields left blank
- Invalid email format
- What happens if the terms are not accepted?
- What happens in a cloud environment vs. OSS? (different sign-ups)

Question, now we have 2 sign-in-page.ts one in /tests/page-objects and one here in this PR, is it right?

StylusFrost committed:
test(ui): update sign-up tests to use E2E_NEW_PASSWORD environment variable

- Modified the sign-up test to retrieve the password from the E2E_NEW_PASSWORD environment variable.
- Updated documentation to specify the requirement for the E2E_NEW_PASSWORD variable before running tests.

StylusFrost committed:
test(ui): update locators in tests to use role-based queries

- Replaced text-based locators with role-based queries in various test files for improved accessibility and consistency.
- Removed deprecated HomePage and SignInPage files to streamline the test structure.

StylusFrost committed:
test(ui): update locators in tests to use role-based queries

- Refactored locators in BasePage, HomePage, SignInPage, and SignUpPage to utilize role-based queries for improved accessibility and consistency across the UI tests.

StylusFrost committed:
test(ui): remove unnecessary blank line in admin authentication setup test

- Cleaned up the admin authentication setup test by removing an unnecessary blank line for improved readability.

jfagoagas committed:
Merge branch 'master' of github.com:prowler-cloud/prowler into PROWLER-187-create-new-user
